modules/targetanalysis/src/target_loader.cpp

```diff
@@ -66,61 +73,63 @@ using namespace std;
 * }
 */
 
-/*
-class targetLoader{
-      targetLoader(string f){
-             file = f;     
-      }
-      public:
-             TargetParameters getParameters();
-             class TargetParameters{
-                   public:
-                          string* parametersTypes;
-                          Parameter* parameters;
-                          unsigned int parameterCount;
-                          class Parameter{
-                                public:
-                                       bool enabled;
-                                       double value;
-                                       unsigned int confidence;
-                          }
-             }
-      private:
-              string file;
-              readFile(string fileLocation);
-              
-}
-
-
-
 
+/*
 TargetParameters targetLoader:getParameters(){
        return 0;
 }
-
-char* targetLoader:readFile(string fileLocation){
+*/
+bool TargetLoader::readFile(string fileLocation, char* data){
      ifstream jFile;
-     char* memblock;
      
      jFile.open(fileLocation, ios::binary | ios::ate);
```

Comment:
The target_compile_features ensures that support is added for the given features, which since most c++ compilers don't use c++11 by default, requiring that it be enabled by a flag, means that since range_for requires c++11, specifying that switches the standard to c++11 (as well as ensuring that the compiler supports the feature).
Note that it's rather confusing to specify that you want range_for to enable a completely unrelated c++11 feature. Honestly you should just convert the string to a c string, though if you wanted to keep c++11, you should either find the feature that relates to ifstream::open taking a string (list [here](https://cmake.org/cmake/help/v3.1/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html)) or, failing that, enable the standard using the CXX_STANDARD cmake property.